crates/gitbutler-branch-actions/src/remote.rs

```diff
@@ -110,6 +110,15 @@ pub(crate) fn get_branch_data(ctx: &CommandContext, refname: &Refname) -> Result
         .context("failed to get branch data")
 }
 
+pub(crate) fn get_commit_data(ctx: &CommandContext, sha: git2::Oid) -> Result<RemoteCommit> {
+    let commit = ctx
+        .repository()
+        .find_commit(sha)
+        .context("failed to find commit")?;
+
+    Ok(commit_to_remote_commit(&commit))
+}
```

Comment:
```suggestion
pub(crate) fn get_commit_data(
    ctx: &CommandContext,
    sha: git2::Oid,
) -> Result<Option<RemoteCommit>> {
    let commit = match ctx.repository().find_commit(sha) {
        Ok(commit) => commit,
        Err(error) => {
            if error.code() == git2::ErrorCode::NotFound {
                return Ok(None);
            } else {
                bail!(error);
            }
        }
    };

    Ok(Some(commit_to_remote_commit(&commit)))
}
```

Reply:
Whenever you see this pattern, that an `err` is conditionally turned into an OK or is kept as the original (or altered) error, there is `.or_else(|err| …)` for you.